### PR TITLE
feat(FX-4535): mark specific notification as read when clicking on an entry in the activity panel

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feat**

This PR solves [FX-4535]

Review app: [mark-notification-as-read](https://mark-notification-as-read.artsy.net/)

### Demo
https://user-images.githubusercontent.com/3513494/213193037-3bdd51be-1f56-4058-a9d6-64c7c6d3cfe9.mp4

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FX-4535]: https://artsyproduct.atlassian.net/browse/FX-4535?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dblandin — APPROVED** [review]
LGTM!

Non-blocking but I think it would be wise to create an upkeep ticket to setup some more robust tests that allow to us exercise some of the tap/click behavior in the activity panel UI.

Hopefully [these docs](https://github.com/artsy/force/blob/main/src/DevTools/README.md#guide-to-createtestenv) are still up-to-date. Would be nice to protect ourselves against regressions by simulating a tap on one of these notification items and ensuring the we're marking as read _and_ navigating to the correct place.

**Comment by dimatretyak** [comment]
> Non-blocking but I think it would be wise to create an upkeep ticket to setup some more robust tests that allow to us exercise some of the tap/click behavior in the activity panel UI

Yes, it's a good idea. I originally planned to add tests, but due to some limitations of how we mock relay env for tests, this could not be done.

**Review by gkartalis — APPROVED** [review]

**Review by nickskalkin — APPROVED** [review]

File: src/Components/Notifications/NotificationItem.tsx [hunk 1]
```diff
@@ -31,16 +37,39 @@ const NotificationItem: React.FC<NotificationItemProps> = ({ item }) => {
...
+        item.internalID
+      )
+      const responseOrError = response.markNotificationAsRead?.responseOrError
+      const errorMessage = responseOrError?.mutationError?.message
+
+      if (errorMessage) {
+        throw new Error(errorMessage)
+      }
+    } catch (error) {
+      logger.error(error)
```
Comment (nickskalkin) [inline src/Components/Notifications/NotificationItem.tsx:58]: **question / non-blocking**: What is our policy for handling exceptions / errors of frontend side? Should we send an error to Sentry here?

Comment (dimatretyak) [reply src/Components/Notifications/NotificationItem.tsx:58]: when we call `logger.error', the error will be [logged in sentry](https://github.com/artsy/force/blob/80d11482e601df71506dac918ea8702d816eb0b7/src/Utils/logger.ts#L23-L25)

**Pull request merged**